am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/RevokeAccessIntegrationTest.java

```diff
@@ -118,6 +119,64 @@ void whenRevokingResourceAccessThatDoesNotExistNoErrorExpected() {
         assertThat(databaseHelper.countExplicitPermissions(resourceId)).isEqualTo(0);
     }
 
+    @Test
+    void whenRevokingResourceWithNonExistentRelationshipShouldNotRemoveAnyRecordFromDatabase() {
+        grantExplicitResourceAccess(resourceId, "");
+
+        service.revokeResourceAccess(ExplicitAccessMetadata.builder()
+            .resourceId(resourceId)
+            .accessorId(accessorId)
+            .accessorType(USER)
+            .serviceName(SERVICE_NAME)
+            .resourceType(RESOURCE_TYPE)
+            .resourceName(RESOURCE_NAME)
+            .attribute(ROOT_ATTRIBUTE)
+            .securityClassification(PUBLIC)
+            .relationship("NonExistentRelationship")
+            .build());
+
+        assertThat(databaseHelper.countExplicitPermissions(resourceId)).isEqualTo(1);
+    }
+
+    @Test
+    void whenRevokingResourceWithNullRelationshipShouldRemoveAnyRecordFromDatabase() {
+        grantExplicitResourceAccess(resourceId, "");
+
+        service.revokeResourceAccess(ExplicitAccessMetadata.builder()
+            .resourceId(resourceId)
+            .accessorId(accessorId)
+            .accessorType(USER)
+            .serviceName(SERVICE_NAME)
+            .resourceType(RESOURCE_TYPE)
+            .resourceName(RESOURCE_NAME)
+            .attribute(JsonPointer.valueOf(""))
+            .securityClassification(PUBLIC)
+            .relationship(null)
+            .build());
+
+        assertThat(databaseHelper.countExplicitPermissions(resourceId)).isEqualTo(0);
+    }
+
+    @Test
+    void whenRevokingResourceWithRelationshipAndParentShouldRemoveParentAndChildrenFromDatabase() {
```

Comment:
That seems like a duplicate of one of already existing cases e.g. `whenRevokingAccessOnAttributeShouldRemoveOnlySpecifiedAttributeAndChildren`

Reply:
Yes it is sorry was testing all scenarios from use cases will remove now.